yarn/src/test/scala/org/apache/spark/deploy/yarn/YarnClusterSuite.scala

```diff
@@ -159,8 +170,19 @@ class YarnClusterSuite extends FunSuite with BeforeAndAfterAll with Matchers wit
 
 }
 
+private class SaveExecutorInfo extends SparkListener {
+  val addedExecutorInfos = mutable.Map[String, ExecutorInfo]()
+
+  override def onExecutorAdded(executor : SparkListenerExecutorAdded) {
+    addedExecutorInfos(executor.executorId) = executor.executorInfo
+  }
+}
+
 private object YarnClusterDriver extends Logging with Matchers {
 
+  val WAIT_TIMEOUT_MILLIS = 10000
+  val listener = new SaveExecutorInfo
```

Comment:
This listener object will be re-used across multiple test runs, which probably isn't ideal in terms of test isolation.